File: BackEnd/Zalo_mini/Zalo_mini/Models/message.cs
using System;
using System.Collections.Generic;
using Zalo_mini.Constants;

namespace Zalo_mini.Models
{
    public partial class message
    {
        public message()
        {
            Inversereply_to_message = new HashSet<message>();
            conversations = new HashSet<conversation>();
            message_deletions = new HashSet<message_deletion>();
            message_reactions = new HashSet<message_reaction>();
            message_read_statuses = new HashSet<message_read_status>();
        }

        public long id { get; set; }
        public long conversation_id { get; set; }
        public long sender_id { get; set; }
        public long? reply_to_message_id { get; set; }
        public string? message_type { get; set; } = MessageType.Text;
        public string? content { get; set; }
        public string? media_url { get; set; }
        public string? file_name { get; set; }
        public long? file_size { get; set; }
        /// <summary>
        /// Duration in seconds for audio/video
        /// </summary>
        public int? duration { get; set; }
        public bool? is_deleted { get; set; }
        public bool? is_recalled { get; set; }
        public bool? is_pinned { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }

        public virtual conversation conversation { get; set; } = null!;
        public virtual message? reply_to_message { get; set; }
        public virtual user sender { get; set; } = null!;
        public virtual ICollection<message> Inversereply_to_message { get; set; }
        public virtual ICollection<conversation> conversations { get; set; }
        public virtual ICollection<message_deletion> message_deletions { get; set; }
        public virtual ICollection<message_reaction> message_reactions { get; set; }
        public virtual ICollection<message_read_status> message_read_statuses { get; set; }
    }
}

File: BackEnd/java/ZaloMini/src/main/java/com/devmam/zalomini/entities/Message.java
package com.devmam.zalomini.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.Instant;
import java.util.Date;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity(name = "Message")
@Table(name = "messages")
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "conversation_id", nullable = false)
    private Conversation conversation;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "sender_id", nullable = false)
    private User sender;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reply_to_message_id")
    private Message replyToMessage;

    @ColumnDefault("'text'")
    @Lob
    @Column(name = "message_type")
    private String messageType;

    @Lob
    @Column(name = "content")
    private String content;

    @Lob
    @Column(name = "media_url")
    private String mediaUrl;

    @Column(name = "file_name")
    private String fileName;

    @Column(name = "file_size")
    private Long fileSize;

    @Column(name = "duration")
    private Integer duration;

    @ColumnDefault("0")
    @Column(name = "is_deleted")
    private Boolean isDeleted;

    @ColumnDefault("0")
    @Column(name = "is_recalled")
    private Boolean isRecalled;

    @ColumnDefault("0")
    @Column(name = "is_pinned")
    private Boolean isPinned;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "updated_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt;

}